src/commands/update.ts

```diff
@@ -49,26 +52,36 @@ export default class UpdateCommand extends Command {
     version: Flags.string({
       char: 'v',
       description: 'Install a specific version.',
-      exclusive: ['interactive'],
+      exclusive: ['interactive', 'force'],
     }),
   }
 
   async run(): Promise<void> {
     const {args, flags} = await this.parse(UpdateCommand)
     const updater = new Updater(this.config)
     if (flags.available) {
-      const [index, localVersions] = await Promise.all([updater.fetchVersionIndex(), updater.findLocalVersions()])
+      const {distTags, index, localVersions} = await lookupVersions(updater, this.config)
+
+      const headers = [
+        {align: 'left', value: 'Location'},
+        {align: 'left', value: 'Version'},
+      ]
+
+      if (distTags) {
+        headers.push({align: 'left', value: 'Channel'})
+      }
 
       // eslint-disable-next-line new-cap
       const t = TtyTable(
-        [
-          {align: 'left', value: 'Location'},
-          {align: 'left', value: 'Version'},
-        ],
+        headers,
         sort(Object.keys(index))
           .reverse()
           .map((version) => {
             const location = localVersions.find((l) => basename(l).startsWith(version)) || index[version]
+            if (distTags) {
+              return [location, version, distTags[version] ?? '']
```

Comment:
> distTags[version]

So in order for this to work, then one must create an NPM tag together with publish to S3?

A bit curious, if the requirement to keep two sources of truth as opposed to just S3. Or S3 doesn't have this kind of information?

Sorry if the question is trivial 🙏 Thank you!

Reply:
Yes, it assumes that you're publishing to both npm and S3. But this assumption has been true for a long time [here](https://github.com/oclif/plugin-update/blob/4.4.0/src/update.ts) and can be turned off by setting `disableNpmLookup` to true in `oclif.update` in your package.json

Really the only way to get the information from S3 is to know ahead of time which channels you want (e.g. I want to show `stable`, `stable-rc`, and `nightly`). But that's hard to do in a plugin like this which is meant to be used by a wide variety of CLIs. For instance one CLI might make heavy use of `nightly` channel whereas another CLI might use a `beta` channel. Of course, this could be configured at the CLI level but I figured that since we were already assuming elsewhere that the CLI is published both to npm and S3 we could continue making that assumption.